ingest/usecase/plugins/orderbookfiller/ordebook_filler_ingest_plugin.go

```diff
@@ -38,11 +43,86 @@ func New(poolsUseCase mvc.PoolsUsecase, routerUseCase mvc.RouterUsecase, tokensU
 // ProcessEndBlock implements domain.EndBlockProcessPlugin.
 func (o *orderbookFillerIngestPlugin) ProcessEndBlock(ctx context.Context, blockHeight uint64, metadata domain.BlockPoolMetadata) error {
 
+	canonicalOrderbooks, err := o.poolsUseCase.GetAllCanonicalOrderbookPoolIDs()
+	if err != nil {
+		o.logger.Error("failed to get all canonical orderbook pool IDs", zap.Error(err))
+		return err
+	}
+
+	resultChan := make(chan error, len(canonicalOrderbooks))
+	defer close(resultChan)
+
+	for _, canonicalOrderbook := range canonicalOrderbooks {
+		go func(canonicalOrderbook domain.CanonicalOrderBooksResult) {
+			resultChan <- o.processOrderbook(ctx, canonicalOrderbook)
+		}(canonicalOrderbook)
+	}
+
+	// Collect all the results
+	for i := 0; i < len(canonicalOrderbooks); i++ {
+		select {
+		case err := <-resultChan:
+			if err != nil {
+				o.logger.Error("failed to process orderbook", zap.Error(err))
+			}
+		case <-time.After(10 * time.Second):
+			o.logger.Error("timed out processing orderbook")
+			return err
+		}
+	}
+
+	o.logger.Info("processed end block in orderbook filler ingest plugin", zap.Uint64("block_height", blockHeight))
+	return nil
+}
+
+func (o *orderbookFillerIngestPlugin) processOrderbook(ctx context.Context, canonicalOrderbookResult domain.CanonicalOrderBooksResult) error {
+
+	baseDenom := canonicalOrderbookResult.Base
+	quoteDenom := canonicalOrderbookResult.Quote
+
+	prices, err := o.tokensUseCase.GetPrices(ctx, []string{baseDenom, quoteDenom}, []string{o.defaultQuoteDenom}, domain.ChainPricingSourceType)
+	if err != nil {
+		return err
+	}
+
+	baseDenomPrice := prices.GetPriceForDenom(baseDenom, o.defaultQuoteDenom)
+	quoteDenomPrice := prices.GetPriceForDenom(quoteDenom, o.defaultQuoteDenom)
+
+	// TODO: make it $10 in USDC terms by using prices above
+	baseInCoin := sdk.NewCoin(baseDenom, sdk.NewInt(10_000_000))
+	baseInOrderbookQuote, err := o.routerUseCase.GetCustomDirectQuote(ctx, baseInCoin, quoteDenom, canonicalOrderbookResult.PoolID)
+	if err != nil {
+		return err
+	}
+
+	// Get execution price
+	baseOrderbookAmount := baseInOrderbookQuote.GetAmountIn()
+	quoteOrderbookAmount := baseInOrderbookQuote.GetAmountOut()
+
+	orderbookQuoteExecutionPrice := baseOrderbookAmount.Amount.ToLegacyDec().Quo(quoteOrderbookAmount.ToLegacyDec())
+
+	// TODO: make it $10 in USDC terms by using prices above
+	quoteInCoin := sdk.NewCoin(quoteDenom, sdk.NewInt(10_000_000))
+
+	quoteInOptimalQuote, err := o.routerUseCase.GetSimpleQuote(ctx, quoteInCoin, baseDenom)
+	if err != nil {
+		return err
+	}
+	baseOptimalQuote := quoteInOptimalQuote.GetAmountOut()
+	quoteOptimalQuote := quoteInOptimalQuote.GetAmountIn()
+
+	// Get execution price
+	optimalQuoteExecutionPrice := baseOptimalQuote.ToLegacyDec().Quo(quoteOptimalQuote.Amount.ToLegacyDec())
+
+	errTolerance := osmomath.ErrTolerance{
+		MultiplicativeTolerance: sdk.NewDecWithPrec(5, 2),
```

Comment:
Is this enough to account for error in `optimalQuoteExecutionPrice` due to e.g. not factoring in gas fee estimates?

Reply:
Will try increasing but Adam's also added logic to account for gas fee